formatter/format_test.go

```diff
@@ -334,6 +334,50 @@ func TestCompilationUnit(t *testing.T) {
 	}
 }
 
+func TestEndOfFileComments(t *testing.T) {
+	if testing.Verbose() {
+		log.SetLevel(log.DebugLevel)
+
+	}
+	tests :=
+		[]struct {
+			input  string
+			output string
+		}{
+			{
+				`private class T1Exception extends Exception {} //test`,
+				`private class T1Exception extends Exception {}
+//test`},
+			{
+				`public class MyClass { public static void noop() {}}
+				//test comment
+				// oie`,
+				`public class MyClass {
+	public static void noop() {}
+}
+//test comment
+// oie`},
```

Comment:
Let's fix this as well, when there's a token after the comment, but no more parse nodes:

```
public class MyClass {
	public static void noop() {}
   // No more nodes after this
}
```